clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/OnDemandAgent.java

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
I turned this to Java because for whatever static classes nested in interfaces in groovy aren't accessible in java

Reply:
Even with `public` keywords? Java port is fine with me just curious.

Reply:
Yup tried all combinations, neither intellij nor gradle would pick itup